tests/unit/DateUtilsTest.js

```diff
@@ -130,6 +130,34 @@ describe('DateUtils', () => {
         expect(result).toBe(expectedDateTime);
     });
 
+    describe('Date Comparison Functions', () => {
+        const today = new Date();
+        const tomorrow = addDays(today, 1);
+        const yesterday = subDays(today, 1);
+
+        const todayInTimezone = utcToZonedTime(today, timezone);
+        const tomorrowInTimezone = utcToZonedTime(tomorrow, timezone);
+        const yesterdayInTimezone = utcToZonedTime(yesterday, timezone);
+
+        it('isToday should correctly identify today', () => {
+            expect(DateUtils.isToday(todayInTimezone, timezone)).toBe(true);
+            expect(DateUtils.isToday(tomorrowInTimezone, timezone)).toBe(false);
+            expect(DateUtils.isToday(yesterdayInTimezone, timezone)).toBe(false);
```

Comment:
FYI it looks like this is failing in early timezones like mine - see https://expensify.slack.com/archives/C01GTK53T8Q/p1699255028952729?thread_ts=1699253289.471629&cid=C01GTK53T8Q

This isn't a HUGE problem since most of our devs are in later timezones, but it's a bit annoying :D

Reply:
Let me know if these changes make sense to you @abzokhattab  - they are currently passing for me: https://github.com/Expensify/App/pull/30900

I'll run these tests again in 20 minutes, when Los Angeles becomes the same day as me :D

Reply:
Interesting!

i still don't get the difference between the two approaches, aren't they give the same result? no?